src/hooks/useSummaryData.ts

```diff
@@ -51,19 +52,34 @@ const useSummaryData = (data: ReportInterface[]) => {
 
   const getKoreanUnit = (value: number): string => {
     const length = value.toFixed(0).length;
-    const regex = /\B(?<!\.\d*)(?=(\d{3})+(?!\d))/g;
+    const initNumberFormat = {
+      number: value,
+      decimalPoint: 0,
+      unit: '',
+    };
+    let formattedValue = '';
 
     if (length < 5) {
       // 1 단위
-      return value.toString().replace(regex, ',');
+      formattedValue = putCommaIntoNumber({ ...initNumberFormat });
     } else if (length < 9) {
       // 만 단위
-      const formattedValue = (value / 10 ** 4).toFixed(0).replace(regex, ',');
-      return `${formattedValue}만`;
+      formattedValue = putCommaIntoNumber({
+        ...initNumberFormat,
+        number: value / 10 ** 4,
+        unit: '만',
+      });
+    } else {
+      // 억 단위
+      formattedValue = putCommaIntoNumber({
+        ...initNumberFormat,
+        number: value / 10 ** 8,
+        decimalPoint: 1,
+        unit: '억',
+      });
```

Comment:
initNumberFormat이 기본 값 설정해주는거라면 요기에는 필요없지 않을까요?
```suggestion
      formattedValue = putCommaIntoNumber({
        number: value / 10 ** 4,
        unit: '만',
      });
    } else {
      // 억 단위
      formattedValue = putCommaIntoNumber({
        number: value / 10 ** 8,
        decimalPoint: 1,
        unit: '억',
      });
```

Reply:
지정 범위에서 2번째 경우는 디스럭쳐링 안해도 괜찮지만 1번째 경우에는 `putCommaIntoNumber` 매개변수 타입에 걸리기 때문에 필요합니다! 2번째는 말씀하신대로 init빼는게 좋겠군요~